graylog2-rest-client/src/main/java/org/graylog2/restclient/lib/ApiClientImpl.java

```diff
@@ -725,5 +730,83 @@ public URL prepareUrl(ClusterEntity target) {
                 throw new RuntimeException(e);
             }
         }
+
+        @Override
+        public InputStream executeStreaming() throws APIException, IOException {
+            if (radio != null && (node != null || nodes != null)) {
+                throw new RuntimeException("You set both and a Node and a Radio as target. This is not possible.");
+            }
+
+            final ClusterEntity target;
+
+            if (radio == null) {
+                if (node == null) {
+                    if (nodes != null) {
+                        LOG.error("Multiple nodes are set, but execute() was called. This is most likely a bug and you meant to call executeOnAll()!", new Throwable());
+                    }
+                    node(serverNodes.any());
+                }
+
+                target = node;
+            } else {
+                target = radio;
+            }
+
+            ensureAuthentication();
+            final URL url = prepareUrl(target);
+            final AsyncHttpClient.BoundRequestBuilder requestBuilder = requestBuilderForUrl(url);
+            requestBuilder.addHeader(HttpHeaders.ACCEPT, mediaType.toString());
+
+            final Request request = requestBuilder.build();
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("API Request: {}", request.toString());
+            }
+
+            // Set 200 OK as standard if not defined.
+            ensureExpectedResponseCodes();
+
+            try {
+                final AsyncByteBufferInputStream stream = new AsyncByteBufferInputStream();
+                requestBuilder.execute(new TransferCompletionHandler().addTransferListener(new TransferListener() {
+                    @Override
+                    public void onRequestHeadersSent(FluentCaseInsensitiveStringsMap headers) {
+                    }
+
+                    @Override
+                    public void onResponseHeadersReceived(FluentCaseInsensitiveStringsMap headers) {
+                        target.touch();
+                    }
+
+                    @Override
+                    public void onBytesReceived(ByteBuffer buffer) throws IOException {
+                        stream.putBuffer(buffer);
+                    }
+
+                    @Override
+                    public void onBytesSent(ByteBuffer buffer) {
+
+                    }
+
+                    @Override
+                    public void onRequestResponseCompleted() {
+                        stream.setDone(true);
+                    }
+
+                    @Override
+                    public void onThrowable(Throwable t) {
+                        stream.setFailed(t);
+                    }
+                }));
+                return stream;
+            } catch (MalformedURLException e) {
+                LOG.error("Malformed URL", e);
```

Comment:
I'm not a 100% sure how the caller would handle the `RuntimeException` but logging this additionally to (re-)throwing the wrapped `MalformedURLException` might lead to excessive* log messages.

*: For values as high as 2…